client/overview/task-list/tasks/update-business-details-task.tsx

```diff
@@ -0,0 +1,120 @@
+ */
+import React from 'react';
+import { __, sprintf } from '@wordpress/i18n';
+import { render } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type { TaskItemProps } from '../types';
+import UpdateBusinessDetailsModal from 'wcpay/overview/modal/update-business-details';
+import { dateI18n } from '@wordpress/date';
+import moment from 'moment';
+
+export const getUpdateBusinessDetailsTask = (
+	errorMessages: string[],
+	status: string,
+	accountLink: string,
+	currentDeadline: number | null,
+	pastDue: boolean
+): TaskItemProps | null => {
+	const accountRestrictedSoon = 'restricted_soon' === status;
+	const accountDetailsPastDue = 'restricted' === status && pastDue;
+	const hasMultipleErrors = 1 < errorMessages.length;
+	const hasSingleError = 1 === errorMessages.length;
+
+	let accountDetailsTaskDescription = '',
+		errorMessageDescription,
+		accountDetailsUpdateByDescription;
+
+	if ( accountRestrictedSoon && currentDeadline ) {
+		accountDetailsUpdateByDescription = sprintf(
+			/* translators: %s - formatted requirements current deadline (date) */
+			__(
+				'Update by %s to avoid a disruption in deposits.',
+				'woocommerce-payments'
+			),
+			dateI18n(
+				'ga M j, Y',
+				moment( currentDeadline * 1000 ).toISOString()
+			)
+		);
+
+		if ( hasSingleError ) {
+			errorMessageDescription = errorMessages[ 0 ];
+			accountDetailsTaskDescription = errorMessageDescription.concat(
+				' ',
+				accountDetailsUpdateByDescription
+			);
+		} else {
+			accountDetailsTaskDescription = accountDetailsUpdateByDescription;
+		}
+	} else if ( accountDetailsPastDue ) {
+		if ( hasSingleError ) {
+			accountDetailsTaskDescription = errorMessages[ 0 ];
+		} else {
+			accountDetailsTaskDescription =
+				/* translators: <a> - dashboard login URL */
+				__(
+					'Payments and deposits are disabled for this account until missing business information is updated.',
+					'woocommerce-payments'
+				);
+		}
+	}
+
+	const renderModal = () => {
+		let container = document.querySelector(
+			'#wcpay-update-business-details-container'
+		);
+
+		if ( ! container ) {
+			container = document.createElement( 'div' );
+			container.id = 'wcpay-update-business-details-container';
+			document.body.appendChild( container );
+		}
+
+		render(
+			<UpdateBusinessDetailsModal
+				key={ Date.now() }
+				errorMessages={ errorMessages }
+				accountStatus={ status }
+				accountLink={ accountLink }
+				currentDeadline={ currentDeadline }
+			/>,
+			container
+		);
+	};
+
+	const handleClick = () => {
+		if ( 'complete' === status ) {
+			return;
+		}
+
+		if ( hasMultipleErrors ) {
+			renderModal();
+		} else {
+			window.open( accountLink, '_blank' );
+		}
+	};
+
+	return {
+		key: 'update-business-details',
+		level: 1,
+		title: __(
+			'Update WooCommerce Payments business details',
+			'woocommerce-payments'
+		),
+		content: accountDetailsTaskDescription,
+		completed: 'complete' === status,
+		onClick: handleClick,
```

Comment:
I wonder why we are also handling redirect on click, but in disputes and reconnect wpcom we are not? 🙂

Reply:
I didn't realise about the Woo Home use case you mentioned, I think I was originally planning to just use `action` instead of `onClick` on all of them (and missed the update on this one).

Now that I know we need the `onClick`, I fixed it in 5e3cfaff3. 🙂

Reply:
oops, I thought I deleted this comment since I've figured it out :D